temperature/router.py

```diff
@@ -0,0 +1,84 @@
+import asyncio
+import os
+from datetime import datetime
+
+import httpx
+from dotenv import load_dotenv
+from fastapi import APIRouter, HTTPException
+
+from city.crud import get_all_cities
+from dependencies import DBSessionDep, PaginationDep
+from temperature import crud, schemas
+
+
+load_dotenv()
+API_KEY = os.getenv("WEATHER_API_KEY")
```

Comment:
Ensure that the `WEATHER_API_KEY` environment variable is set correctly in your deployment environment. If it's missing, the `fetch_weather` function will raise a `ValueError`.